src/main/java/com/google/api/codegen/transformer/DiscoGapicMethodContext.java

```diff
@@ -96,4 +95,19 @@ public DiscoGapicMethodContext cloneWithEmptyTypeTable() {
 
   @Override
   public abstract SchemaTypeTable getTypeTable();
+
+  @Override
+  public String getAndSaveRequestTypeName() {
+    return getTypeTable().getAndSaveNicknameFor(getMethod().request());
+  }
+
+  @Override
+  public String getAndSaveResponseTypeName() {
+    return getTypeTable().getAndSaveNicknameFor(getMethod().response());
+  }
+
+  @Override
+  public String getBatchingDescriptorConstName() {
+    return getNamer().getBatchingDescriptorConstName(getMethod());
+  }
```

Comment:
I promise promise to remove this in a future PR! I have another work-in-progress PR that makes a shared interface for Method models that would solve this problem, but that's a large enough change that warrants its own review.

There's now a TODO for it.